lib/wallet/http.js

```diff
@@ -1246,6 +1247,83 @@ class HTTP extends Server {
       return res.json(200, mtx.getJSON(this.network));
     });
 
+    // Batch Bid
+    this.post('/wallet/:id/batch/bid', async (req, res) => {
+      const MAX_BIDS_ARRAY_LENGTH = 200;
+      const valid = Validator.fromRequest(req);
+      const passphrase = valid.str('passphrase');
+
+      if (!valid.has('bids'))
+         throw new Error('Mandatory Bids parameter is missing!');
+
+      let bids = valid.array('bids');
+      assert(bids.length > 0 && bids.length <= MAX_BIDS_ARRAY_LENGTH,
+        `Bids are required to be non empty and shoud not exceed 
+        ${MAX_BIDS_ARRAY_LENGTH}`);
+
+      bids.forEach((element) => {
+        const validator = new Validator(element);
+        if (!validator.has('name') ||
+            !validator.has('bid')  ||
+            !validator.has('lockup') ||
+            !validator.has('idempotencyKey')) {
+          throw new Error('Bids contains a Bid with missing mandatory fields!');
+        } else { // check types
+          validator.str('name');
+          validator.u64('bid');
+          validator.u64('lockup');
+          validator.str('idempotencyKey');
+        }
+      });
+
+     // add value property (same as bid) to each bid
+     // value property is same with bid property and used internally
+     bids = bids.map((element) => {
+        element['value'] = element.bid;
+        return element;
+      });
+
+      const options = TransactionOptions.fromValidator(valid);
+
+      // if already exists in the cache do not process it
+      let results = [];
+      let errors  = [];
+
+      const filteredBids = bids.filter((bid) => {
+          const bidResultFromCache = req.wallet
+          .sendBatchBidResults.get(bid.idempotencyKey);
```

Comment:
Is there a reason we can't use the same bid cache for batched/non batched bids?

Reply:
no reason as far as i can see, let me update.

Reply:
fixed ...